qcodes/dataset/measurements.py

```diff
@@ -159,23 +181,34 @@ def add_result(self,
                                      f'setpoint values for {param}:'
                                      f' {stuffweneed}.'
                                      f' Values only given for {params}.')
+        if inserting_unrolled_array and inserting_as_arrays:
+            raise RuntimeError("Trying to insert multiple data values both "
+                               "in array from and as numeric. This is not "
+                               "possible.")
+        elif inserting_as_arrays:
+            input_size = 1
 
         for index in range(input_size):
             res_dict = {}
             for partial_result in res:
-                param = str(partial_result[0])
-                value = partial_result[1]
-                # For compatibility with the old Loop, setpoints are
-                # tuples of numbers (usually tuple(np.linspace(...))
-                if hasattr(value, '__len__') and not(isinstance(value, str)):
-                    value = cast(Union[Sequence,np.ndarray], value)
-                    if isinstance(value, np.ndarray):
-                        value = np.atleast_1d(value)
-                    res_dict.update({param: value[index]})
-                else:
-                    res_dict.update({param: value})
-
-            self._results.append(res_dict)
+                param_spec = self.parameters[str(partial_result[0])]
+                if param_spec.type == 'array' and index == 0:
+                    res_dict[str(partial_result[0])] = partial_result[1]
+                elif param_spec.type != 'array':
```

Comment:
btw, shall we finally make an enum with these SQLite data types? Isn't there already one in `sqlite` module?

Reply:
Done by exposing the list in ParamSpec